src/App.tsx

```diff
@@ -1,34 +1,116 @@
 /* eslint-disable max-len */
-import React from 'react';
+import { FC, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { Todo, FilterType } from './types';
+import { getTodos } from './api';
 import { TodoList } from './components/TodoList';
-import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { TodoModal } from './components/TodoModal';
+import { TodoFilter } from './components/TodoFilter';
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[] | null>(null);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterType, setFilterType] = useState<FilterType>('all');
+  const [query, setQuery] = useState<string>('');
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      try {
+        const allTodos = await getTodos();
```

Comment:
Your solution might seem slower, because you are awaiting getTodos with each character typed. Instead, fetch todos once, store them in the state and filter them ;)
This should already improve the performance. If not, then you can try to add some hooks like `useMemo` or something else ;)